test/browser/suspense.test.js

```diff
@@ -0,0 +1,354 @@
+/*eslint-env browser, mocha */
+/** @jsx h */
+import { setupRerender } from 'preact/test-utils';
+import { createElement as h, render, Component, Suspense, lazy, Fragment } from '../../src/index';
+import { setupScratch, teardown } from '../_util/helpers';
+
+class LazyComp extends Component {
+	render() {
+		return <div>Hello from LazyComp</div>;
+	}
+}
+
+function CustomSuspense({ isDone, start, name }) {
+	if (!isDone()) {
+		throw start();
+	}
+
+	return (
+		<div>
+			Hello from CustomSuspense {name}
+		</div>
+	);
+}
+
+class Catcher extends Component {
+	constructor(props) {
+		super(props);
+		this.state = { error: false };
+	}
+
+	componentDidCatch(e) {
+		this.setState({ error: e });
+	}
+
+	render(props, state) {
+		return state.error ? <div>Catcher did catch: {state.error.message}</div> : props.children;
+	}
+}
+
+function createSuspension(name, timeout, t) {
+	let done = false;
+	let prom;
+
+	return {
+		name,
+		start: () => {
+			if (!prom) {
+				prom = new Promise((res, rej) => {
+					setTimeout(() => {
+						done = true;
+						if (t) {
+							rej(t);
+						}
+						else {
+							res();
+						}
+					}, timeout);
+				});
+			}
+
+			return prom;
+		},
+		getPromise: () => prom,
+		isDone: () => done
+	};
+}
+
+class WrapperOne extends Component {
+	render() {
+		return this.props.children;
```

Comment:
Can this component render a the children inside of a `div`? I'd like to validate that when suspending any intermediate DOM is removed and the fallback replaces all of it.

Reply:
Done. Also added a function component wrapper